server/container/guice/guice-common/src/main/java/org/apache/james/PeriodicalHealthChecks.java

```diff
@@ -0,0 +1,64 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import java.time.Duration;
+import java.util.Set;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import org.apache.james.core.healthcheck.HealthCheck;
+import org.apache.james.lifecycle.api.Startable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class PeriodicalHealthChecks implements Startable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PeriodicalHealthChecks.class);
+    private static final long INITIAL_DELAY = 1;
+    private static final long PERIOD = 1;
+
+    private final Flux<HealthCheck> healthChecks;
+    private Disposable disposable;
+
+    @Inject
+    PeriodicalHealthChecks(Set<HealthCheck> healthChecks) {
+        this.healthChecks = Flux.fromIterable(healthChecks);
+    }
+
+    public void start() {
+        disposable = Flux.interval(Duration.ofSeconds(INITIAL_DELAY), Duration.ofSeconds(PERIOD))
+            .flatMap(any ->
+                healthChecks.flatMap(healthCheck -> Mono.just(healthCheck.check())))
+            .subscribeOn(Schedulers.elastic())
+            .subscribe();
```

Comment:
do you think it's nicer for administrators to gather all results and report everything at one place in here?

Reply:
Not sure to get what you mean here? Each healthCheck is so far doing a bit on his own, and logging his own health when being called. If you mean he should return a result and do all the logging/report here, is not so easy, would require quite some work to unify all of this, and should be done (if valid) in an other PR?

Reply:
It was the original goal of the ticket. However since each healthcheck is logging their component's status when method check() is called,  logging results here would be overlapping. We could modify the healthchecks, but then the endpoint would be affected. 

May be if we try your idea above about using a common component for healthchecking process, then this also could be achieved. It's up to debate i guess.

Reply:
Back to history, can I ask what is the purpose of this periodically healtcheck? if it is designed to help administrators being aware of James health. Then log files are not so friendly right? I have to dig into james logs to see what happened. I rather put in in grafana board or something else, or mailing maybe...

Reply:
refer to this: https://ci.linagora.com/linagora/lgs/openpaas/james/issues/1621